docs/developer/buddypress-database-diagram.md

```diff
@@ -0,0 +1,388 @@
+                                 +----------------------+
+```
+
+```
+       +---------------------+    +--------------------+
+       | wp_bp_optouts        |    | wp_bp_signups      |
+       +---------------------+    +--------------------+
+       | id                  |    | signup_id          |
+       | email_address_hash   |    | domain             |
+       | user_id             |    | path               |
+       | email_type          |    | title              |
+       | date_modified       |    | user_login         |
+       +---------------------+    | user_email         |
+                                  | registered         |
+                                  | activated          |
+                                  | active             |
+                                  | activation_key     |
+                                  | meta               |
+                                  +--------------------+
+```
+
+## Activity Tables
+
+### wp_bp_activity
+
+This table stores all activities (such as posts, comments, updates) made by users in BuddyPress.
+
+- **Columns:**
+  - `id`: The unique ID for each activity entry.
+  - `user_id`: The ID of the user who performed the activity.
+  - `component`: Specifies which BuddyPress component (e.g., groups, members) the activity is associated with.
+  - `type`: Type of activity (e.g., new post, comment, group creation).
+  - `action`: Human-readable description of the action performed.
+  - `content`: Additional content (text) related to the activity.
+  - `primary_link`: A link to the activity item or content.
+  - `item_id`: ID associated with the primary item (group or other).
+  - `secondary_item_id`: Additional context ID, if applicable.
+  - `date_recorded`: When the activity was recorded.
+  - `hide_sitewide`: Whether the activity is hidden from the sitewide activity stream.
+  - `mptt_left` & `mptt_right`: Used for threaded comments/activities.
+  - `is_spam`: Marks the activity as spam.
+
+#### Example Query:
+
+```sql
+SELECT * FROM wp_bp_activity WHERE user_id = 1;
+```
+
+### wp_bp_activity_meta
+
+Stores metadata associated with activity entries.
+
+- **Columns:**
+  - `id`: Meta ID.
+  - `activity_id`: The ID of the associated activity.
+  - `meta_key`: Metadata key.
+  - `meta_value`: Metadata value (usually stored as serialized data).
+
+---
+
+## Friends Tables
+
+### wp_bp_friends
+
+This table manages the friendship connections between users.
+
+- **Columns:**
+  - `id`: Unique ID for each friendship connection.
+  - `initiator_user_id`: The user who sent the friend request.
+  - `friend_user_id`: The user who received the friend request.
+  - `is_confirmed`: Whether the friendship has been confirmed (1) or is pending (0).
+  - `is_limited`: Whether the friendship has any limitations.
+  - `date_created`: Timestamp of when the friendship was initiated.
+
+#### Example Query:
+
+```sql
+SELECT * FROM wp_bp_friends WHERE initiator_user_id = 1;
+```
+
+---
```

Comment:
It should be in `docs/developer/components/friends/db-schema.md` and use a table like the what I've written for the Activity above.